reference-implementation/to-upstream-wpts/transform-streams/errors.js

```diff
@@ -56,4 +56,125 @@ promise_test(t => {
   ]);
 }, 'TransformStream errors thrown in flush put the writable and readable in an errored state');
 
+async_test(t => {
+  new TransformStream({
+    start(c) {
+      t.step(() => {
+        c.enqueue('a');
+        c.error(new Error('generic error'));
+        assert_throws(new TypeError(), () => c.enqueue('b'), 'Errored TransformStream cannot enqueue new chunks');
+        t.done();
+      });
+    },
+    transform() {
+    }
+  });
+}, 'TransformStream cannot be used after becoming errored');
+
+async_test(t => {
+  const ts = new TransformStream({
+    start() {
+      return new Promise(resolve => setTimeout(resolve, 90))
+        .then(() => { throw thrownError; });
+    },
+    transform() {
+      t.step(() => assert_unreached('transform must never be called if start() fails'));
+    },
+    flush() {
+      t.step(() => assert_unreached('flush must never be called if start() fails'));
+    }
+  });
+
+  const writer = ts.writable.getWriter();
+  writer.write('a').then(t.step_func(() => {
+    assert_unreached('writer should be errored if start() fails');
+  }))
+  .catch(t.step_func(e => assert_equals(e, thrownError, 'writer rejects with same error')));
+  writer.close().then(t.step_func(() => {
+    assert_unreached('writer should be errored if start() fails');
+  }))
+  .catch(t.step_func(e => assert_equals(e, thrownError, 'writer rejects with same error')));
```

Comment:
as we've removed plan(), we need to add some logic to check that this expectation is checked before running t.done()?